dracut/30ignition/ignition-generator

```diff
@@ -39,4 +39,8 @@ if $(cmdline_bool 'ignition.firstboot' 0); then
     add_requires ignition-complete.target
 fi
 
+escaped_guid=$(systemd-escape --path "/dev/disk/by-diskuuid/00000000-0000-0000-0000-000000000001")
+add_requires "disk-uuid@${escaped_guid}.service"
```

Comment:
One thing here is this will make rerunning Ignition "manually" noticeably harder.  I think that's fine, just noting it.

It feels weird to have a templated `@` unit that we only instantiate once.  But eh, also fine.

The `diskuuid` naming for the udev rule seems fine, but given that the systemd unit hardcodes running `sgdisk` and such, perhaps it's better named `ignition-gpt-setup@.service`?

Reply:
I could go either way on the instantiated unit. Might make it easier for other distros using the code in case they want to override it, but that just moves the change to ignition-generator instead of disk-uuid.service. (@Conan-Kudo, @laenion do y'all have a preference?)

+1 to renaming. It'd not strictly for Ignition (after all it _is_ a guid that ought to be unique across machines). Maybe `gpt-setup.service`?)

Reply:
`coreos-gpt-setup.service` since it's part of our umbrella?

Reply:
This is certainly interesting because it ties Ignition to the block device setup.  Hmm...I think actually a cleaner way to handle this would be to inspect the disk image we know we were booted from - what [coreos-growpart](https://github.com/coreos/fedora-coreos-config/blob/testing-devel/overlay/usr/libexec/coreos-growpart) does.

That also makes it much easier to implement idempotency - we only scramble the guid if it isn't already scrambled, which would fix the "reprovision in place" model.

Reply:
Oh hum...but the bootloader loaded the initramfs...so at that point we don't actually know what block device we came from unless the bootloader tells us somehow.  But, we know the *default* target root filesystem; we can wait for the first block device with the `root` label?

Reply:
Getting the bootloader to tell us is also not easy since grub doesn't know how it's devices will map to whatever the kernel names things.

We'd potentially run into issues if someone moved root (once we get that working) then reran ignition, but rerunning ignition is already not supported so I don't think we need to worry about that.

To be clear, you're proposing dropping the udev rule (fine by me), making it depend on /dev/disk/by-[part?]label/root and making it only reset the uuid if its the sentinel? Do you know of an easy way to go from part device to the block device containing that partition?

Reply:
Yeah; to rephrase, we already have an "API" to which the root device is required to conform with the labels (`root`, `boot` etc.). 

And yep, going from the partition to the root device... https://gist.github.com/cgwalters/020e0a671f9fa02d155fcc604dc898b0

Reply:
In the process of working on this I found an even easier way. `lsblk -p -P -o PKNAME,PTUUID part_dev` will print the gpt disk uuid and parent device in NAME=VALUE format, which is perfect.

Reply:
Note: https://src.fedoraproject.org/rpms/ignition/pull-request/41#comment-30084

Not sure whether there are plans to update `util-linux` in el8 or how hard it'd be to backport. Might be easier to just go with Colin's approach instead.